features/include/pcl/features/impl/good.hpp

```diff
@@ -0,0 +1,750 @@
+ *
+ */
+
+#ifndef PCL_FEATURES_IMPL_GOOD_H_
+#define PCL_FEATURES_IMPL_GOOD_H_
+#include <pcl/features/good.h>
+#include <pcl/filters/project_inliers.h>
+
+
+/** \brief GOOD: a Global Orthographic Object Descriptor for 3D object recognition and manipulation.
+  * GOOD descriptor has been designed to be robust, descriptive and efficient to compute and use. 
+  * It has two outstanding characteristics: 
+  * 
+  * (1) Providing a good trade-off among :
+  *	- descriptiveness,
+  *	- robustness,
+  *	- computation time,
+  *	- memory usage.
+  * 
+  * (2) Allowing concurrent object recognition and pose estimation for manipulation.
+  * 
+  * \note This is an implementation of the GOOD descriptor which has been presented in the following papers:
+  * 
+  *	[1] Kasaei, S. Hamidreza,  Ana Maria Tomé, Luís Seabra Lopes, Miguel Oliveira 
+  *	"GOOD: A global orthographic object descriptor for 3D object recognition and manipulation." 
+  *	Pattern Recognition Letters 83 (2016): 312-320.http://dx.doi.org/10.1016/j.patrec.2016.07.006
+  *
+  *	[2] Kasaei, S. Hamidreza, Luís Seabra Lopes, Ana Maria Tomé, Miguel Oliveira 
+  * 	"An orthographic descriptor for 3D object learning and recognition." 
+  *	2016 IEEE/RSJ International Conference on Intelligent Robots and Systems (IROS), Daejeon, 2016, 
+  *	pp. 4158-4163. doi: 10.1109/IROS.2016.7759612
+  * 
+  * Please adequately refer to this work any time this code is being used by citing above papers.
+  * If you do publish a paper where GOOD descriptor helped your research, we encourage you to cite the above papers in your publications.
+  * 
+  * \author Hamidreza Kasaei (Seyed.Hamidreza[at]ua[dot]pt)
+  */
+
+template <typename PointInT>
+pcl::GOODEstimation<PointInT>::GOODEstimation()
+{
+  number_of_bins_ = 5;
+  threshold_ = 0.0015;
+}
+
+template <typename PointInT>
+pcl::GOODEstimation<PointInT>::GOODEstimation(unsigned int number_of_bins, float threshold)
+{
+  number_of_bins_ = number_of_bins;
+  threshold_ = threshold;
+};
+//////////////////////////////////////////////////////////////////////////////////////////////
+    
+/*______________________________
+|                               |
+|           Methods             |
+|_______________________________| */
+
+template <typename PointInT>
+void 
+pcl::GOODEstimation<PointInT>::setInputCloud (boost::shared_ptr<pcl::PointCloud<PointInT> > cloud)
+{ 
+  input_ = cloud; 
+};
+//////////////////////////////////////////////////////////////////////////////////////////////
+
+template <typename PointInT>
+void
+pcl::GOODEstimation<PointInT>::setNumberOfBins (unsigned int number_of_bins)
+{
+  number_of_bins_ = number_of_bins;
+}
+//////////////////////////////////////////////////////////////////////////////////////////////
+
+template <typename PointInT>
+void
+pcl::GOODEstimation<PointInT>::setThreshold (float threshold)
+{
+  threshold_ = threshold;
+}
+//////////////////////////////////////////////////////////////////////////////////////////////
```

Comment:
Move both of these to the .h and define them inline and add the const qualifier to the function argument.